### ISSUE TITLE
Be more clear when attacks need logits or softmax probs

### ISSUE THREAD

**Issue by npapernot** [tensorflow/cleverhans#160]
Some attacks operate with the model's logits and others with the softmax probs. We should improve the documentation to make sure it is clear when the `model` argument of an attack class needs to be logits and when it needs to be softmax probs (see #157). In addition, we could use #154 to have some more automated checks in the `__init__` method of attacks that test whether the `model` argument is appropriate. There could also be some more generic checks looking for the TensorFlow softmax op.

**Comment by npapernot** [comment]
Now that the model abstraction is merged, we should allow a model object to be passed to the attacks to make these things more clear.

Besides that, a pass to the docstrings to check for consistency in the description of what layer is expected to be fed should be sufficient to close this issue.

**Comment by npapernot** [comment]
#199 takes care of the first point

**Comment by carlini** [comment]
Is this done?

**Comment by npapernot** [comment]
Not yet, I started working on it locally and will make a PR this week
